plugins/module_utils/common.py

```diff
@@ -537,30 +526,37 @@ def execute_module(self):
         except (urllib3.exceptions.RequestError) as e:
             self.fail_json(msg="Couldn't connect to Kubernetes: %s" % str(e))
 
-        flattened_definitions = []
-        for definition in self.resource_definitions:
-            if definition is None:
-                continue
-            kind = definition.get('kind', self.kind)
-            api_version = definition.get('apiVersion', self.api_version)
-            if kind and kind.endswith('List'):
-                resource = self.find_resource(kind, api_version, fail=False)
-                flattened_definitions.extend(self.flatten_list_kind(resource, definition))
-            else:
-                resource = self.find_resource(kind, api_version, fail=True)
-                flattened_definitions.append((resource, definition))
-
-        for (resource, definition) in flattened_definitions:
-            kind = definition.get('kind', self.kind)
-            api_version = definition.get('apiVersion', self.api_version)
-            definition = self.set_defaults(resource, definition)
-            self.warnings = []
-            if self.params['validate'] is not None:
-                self.warnings = self.validate(definition)
-            result = self.perform_action(resource, definition)
+        if self.params["merge_type"] and self.params["merge_type"][0] == "json":
```

Comment:
what if ``merge_type`` is defined like this : 
```
- merge
- json
```
according to the documentation 
``If more than one merge_type is given, the merge_types will be tried in order``

Reply:
So the problem here is that `resource_definition` is usually treated as a list of definitions that are processed individually, but a json patch is a list of operations that need to be processed as one. I'm not sure how we support this without a complete rewrite of the module, and even then, I think this is going to be difficult. It's hard for me to think of a good reason why someone would list both json patch and either json merge or strategic merge for `merge_type` as the patch formats are completely different.

I'm open to suggestions for how to address this.